src/senaite/core/content/analysisprofile.py

```diff
@@ -295,6 +298,17 @@ def setServices(self, value):
                 raise TypeError(
                     "Expected object, uid or record, got %r" % type(v))
             records.append({"uid": uid, "hidden": hidden})
+
+        # always keep inactive services so they come up again when reactivated
+        uids = [record.get("uid") for record in records]
+        for record in self.getRawServices():
+            uid = record.get("uid")
+            if uid in uids:
+                continue
+            obj = api.get_object_by_uid(uid)
+            if not api.is_active(obj):
+                records.append(record)
```

Comment:
Edit:
I understand now why this is needed and suggest the same changes as mentioned below. Also I would add the `self.getServiceUIDs` to avoid the record handling in the loop

Reply:
Addressed with https://github.com/senaite/senaite.core/pull/2637/commits/df4bf1432c89c94257ae2cc42f50981eee10d8c2